demo/callbacks.py

```
@@ -0,0 +1,359 @@
+    report: Report
+    data_path: Path
+
+
+def evaluate_demo(data_path: Optional[str] = None) -> Optional[EvaluationResult]:
+    """Load the dataset and run evaluation, returning None on failure."""
+    path = Path(data_path) if data_path else _DEFAULT_DATA_PATH
+    try:
+        pipeline = build_pipeline()
+        dataset = load_dataset(str(path))
+        report = run_evaluation(pipeline, dataset)
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)